src/main/java/org/opensearch/ad/transport/SearchAnomalyResultTransportAction.java

```diff
@@ -105,6 +120,7 @@ protected void doExecute(Task task, SearchRequest request, ActionListener<Search
             try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
                 // Search result indices of all detectors. User may create index with same prefix of custom result index
                 // which not used for AD, so we should avoid searching extra indices which not used by anomaly detectors.
+                boolean finalOnlyQueryCustomResultIndex = onlyQueryCustomResultIndex;
```

Comment:
can you explain why this is needed?

Reply:
sure, will add some comments
"Variable used in lambda expression should be final or effectively final"